addons/mail/wizard/mail_compose_message.py

```diff
@@ -306,8 +306,8 @@ def _action_send_mail(self, auto_commit=False):
                             message_type=wizard.message_type,
                             subtype_id=subtype_id,
                             email_layout_xmlid=wizard.email_layout_xmlid,
-                            email_add_signature=not bool(wizard.template_id) and wizard.email_add_signature,
-                            mail_auto_delete=wizard.template_id.auto_delete if wizard.template_id else self._context.get('mail_auto_delete', True),
+                            email_add_signature=not bool(wizard.template_id.exists()) and wizard.email_add_signature,
+                            mail_auto_delete=wizard.template_id.auto_delete if wizard.template_id.exists() else self._context.get('mail_auto_delete', True),
```

Comment:
Don't do that, it adds queries for no reason. At that point the template should be handled like a classic m2o field.